14_contains/solution/contains-solution.js

```diff
@@ -0,0 +1,16 @@
+const contains = function(obj, searchValue) {
+  const values = Object.values(obj);
+
+  // NaN !== NaN, so we would usually have to do an explicit check with Math.isNaN to test for it. However, Array.prototype.includes handles that for us
```

Comment:
I'm maybe a bit wary of this, but I'll defer to the JS people on it. Generally I'd say code that demands a commented explanation is a bit of a warning sign, but `NaN` behavior is one of those things that's so weird, maybe it'd need a comment no matter how it's written.

Reply:
The `NaN === NaN` returning `false` thing will definitely throw a lot of people off, I can imagine. It's not something people would've come across much by this point but will be a good time to learn about that.

That's why I personally would write a specific NaN check in, though I'm also not opposed to using `.includes`. The comment there is mainly as a teaching point that `.includes` has built-in `NaN` checking in case anyone had to do an explicit NaN check but then saw the provided solution didn't.
I imagine Ruby's `nil` behaves a little more...intuitively?

Reply:
Oh right you are. I was thinking the way it was currently written could handle sibling branches. Nice spot.

Reply:
I commented on the wrong thread ... but you get my point

Reply:
yeah js can be a little counter-intuitive at times